subcommand/consul-sidecar/command.go

```diff
@@ -125,12 +177,60 @@ func (c *Command) Run(args []string) int {
 		// Re-loop after syncPeriod or exit if we receive interrupt or terminate signals.
 		case <-time.After(c.flagSyncPeriod):
 			continue
-		case <-ctx.Done():
+		case <-signalCtx.Done():
 			return 0
 		}
 	}
 }
 
```

Comment:
How to test the merged metrics server? Is it worth running it? It has nowhere realistic to scrape from.

Some interesting behaviors maybe worth testing: if it can't fetch app metrics, it should log and just return envoy metrics in the response. If it can't fetch envoy metrics, it should log and not fetch anything.

Should we make a mock server that either returns "fake metrics" or returns a 404, and 404 simulates no metrics endpoint and "fake metrics" simulates metrics.

Reply:
We could potentially have a mock server that we start when we want to test "fake metrics" but dont start at all when we want to test it returning a `404`

Reply:
Thanks ashwin! I like this idea.